### ISSUE TITLE
TweetTokenizer fails to remove user handles in text.

### ISSUE THREAD

**Issue by 12mohaned** [nltk/nltk#2792]
The below input should remove all user handles which start with "@".

input: @remy:This is waaaaayyyy too much for you!!!!!!@adam 
output : [':', 'This', 'is', 'waaayyy', 'too', 'much', 'for', 'you', '!', '!', '!', '@adam']

The TweetTokenizer fail to remove the user handle of Adam.

I would like to open a pull request that solves the following issues:- 

- Improve the regular expression pattern to detect in-text handles properly.
- Adjust the length of characters to be 15 instead of 20.

**Comment by 404-P4rziv4L** [comment]
Hey @tomaarsen, I would like to work on this if someone hasn't started already.

**Comment by tomaarsen** [comment]
@Abhijnan-Bajpai Feel free! I don't believe someone has started on this yet, and we're always open to PRs. There's some guidance over at [CONTRIBUTING.md](https://github.com/nltk/nltk/blob/develop/CONTRIBUTING.md) that might help you.

**Comment by 404-P4rziv4L** [comment]
> @Abhijnan-Bajpai Feel free! I don't believe someone has started on this yet, and we're always open to PRs. There's some guidance over at [CONTRIBUTING.md](https://github.com/nltk/nltk/blob/develop/CONTRIBUTING.md) that might help you.

Cool, thanks for green signal!

**Comment by tomaarsen** [comment]
@Abhijnan-Bajpai Oh, and in case it helps, the regular expression used for stripping handles can be found here:
https://github.com/nltk/nltk/blob/d4e8c3c0e99de11f23d19cfe060591a7afb738c3/nltk/tokenize/casual.py#L180-L184

The remainder of that file has the TweetTokenizer code.

**Comment by 404-P4rziv4L** [comment]
> @Abhijnan-Bajpai Oh, and in case it helps, the regular expression used for stripping handles can be found here:
> https://github.com/nltk/nltk/blob/d4e8c3c0e99de11f23d19cfe060591a7afb738c3/nltk/tokenize/casual.py#L180-L184
> 
> The remainder of that file has the TweetTokenizer code.

Wanted to know @tomaarsen whether the implementation has to be in the form of a regex? I was thinking of an implementation where I could use findAll to find occurrences of handles and stripe the text.

**Comment by tomaarsen** [comment]
@Abhijnan-Bajpai It doesn't just have to be in the form of a regex. I only linked it so you knew where in the codebase to look for TweetTokenizer. You can implement your changes however you think works best.

**Comment by 404-P4rziv4L** [comment]
Just noticed that the example given in the issue statement doesn't seem to have a space between normal text and the handle mention. In a similar way there could also be cases where there is no space after the handle mention. For such cases how do we differentiate between the normal and handle text? Adding to that, texts where '@' is used as a replacement for 'a' will cause more issues with an implementation which solves the above issue. 
The only way I could think of was to use an API and check the existence of non space-separated handle mentions. If they're users, we could stripe, otherwise its a part of the tweet.

**Comment by tomaarsen** [comment]
> For such cases how do we differentiate between the normal and handle text?

In the example, `@adam` does get split from the remainder of the text already, even without a space. For example:
```python
>>> tok.tokenize("@remy:This is waaaaayyyy too much for you!!!!!!@adam")
[':', 'This', 'is', 'waaaaayyyy', 'too', 'much', 'for', 'you', '!', '!', '!', '@adam']
>>> tok.tokenize("@remy:This is waaaaayyyy too much for you@adam")       
[':', 'This', 'is', 'waaaaayyyy', 'too', 'much', 'for', 'you', '@adam']
```
This provides another possible solution - this list can be post-processed to remove all handles. However, this might not work for all edge cases.

**Comment by 12mohaned** [comment]
I believe the regular expression is made on purpose to ignore username handles with no spaces before the text. In twitter, handles are not highlighted as usernames if there's preceding text before them without spaces. I am not sure but I think I had mistaken for an issue.

**Comment by tomaarsen** [comment]
Ah, I see. So, there's a possibility that there is no bug present at all? Do we know what happens if someone tweets e.g. `!@hello`, would `@hello` be turned into a handle?

**Comment by 404-P4rziv4L** [comment]
@tomaarsen it doesn't, checked up on it.

**Comment by tomaarsen** [comment]
That would suggest that there is no bug, but feel free to correct me if I'm wrong.

**Comment by 404-P4rziv4L** [comment]
I think I should just fix the length to 15 from 20 then

**Comment by tomaarsen** [comment]
Thank you for raising this issue, @12mohaned. I believe it was appropriately resolved in #2799. Feel free to let me know if you disagree.
